src/main/java/org/openstreetmap/atlas/generator/AtlasGenerator.java

```diff
@@ -284,243 +360,428 @@ public void start(final CommandMap command)
-                    {
-                        atlasLoadingOption.setOsmPbfNodeFilter(
-                                getTaggableFilterFrom(pbfNodeConfiguration, configurationMap));
-                    }
-                    if (pbfWayConfiguration != null)
-                    {
-                        atlasLoadingOption.setOsmPbfWayFilter(
-                                getTaggableFilterFrom(pbfWayConfiguration, configurationMap));
-                    }
-                    if (pbfRelationConfiguration != null)
-                    {
-                        atlasLoadingOption.setOsmPbfRelationFilter(
-                                getTaggableFilterFrom(pbfRelationConfiguration, configurationMap));
-                    }
+                    // Set the country code that is being processed!
+                    final AtlasLoadingOption atlasLoadingOption = buildAtlasLoadingOption(
+                            boundaries, sparkContext, loadingOptions);
+                    atlasLoadingOption.setAdditionalCountryCodes(countryName);
 
-                    // Build the appropriate PbfLoader
+                    // Build the PbfLoader
                     final PbfLoader loader = new PbfLoader(pbfContext, sparkContext, boundaries,
                             atlasLoadingOption, codeVersion, dataVersion, task.getAllShards());
                     final String name = countryName + CountryShard.COUNTRY_SHARD_SEPARATOR
                             + shard.getName();
                     final Atlas atlas;
                     try
                     {
-                        // Generate the Atlas for this shard
-                        atlas = loader.load(countryName, shard);
+                        // Generate the raw Atlas for this shard
+                        atlas = loader.generateRawAtlas(countryName, shard);
                     }
                     catch (final Throwable e)
                     {
-                        throw new CoreException("Building Atlas {} failed!", name, e);
+                        throw new CoreException("Building raw Atlas {} failed!", name, e);
                     }
 
                     // Report on memory usage
-                    logger.info("Printing memory after loading Atlas {}", name);
+                    logger.info("Printing memory after loading raw Atlas {}", name);
                     Memory.printCurrentMemory();
+
                     // Output the Name/Atlas couple
                     final Tuple2<String, Atlas> result = new Tuple2<>(name, atlas);
                     return result;
-                });
+                }).filter(tuple -> tuple._2() != null);
 
-        // Filter out null Atlas.
-        final JavaPairRDD<String, Atlas> countryNonNullAtlasShardsRDD = countryAtlasShardsRDD
-                .filter(tuple -> tuple._2() != null);
+        // Persist the RDD to avoid re-computing and save each raw atlas
+        countryRawAtlasShardsRDD.cache().saveAsHadoopFile(
```

Comment:
Same, bubble this up at the top level. `cache()` calls are integral to understanding the Spark flow, and I would love to see them in the top method..